pkg/protocol/protocol_fork.go

```diff
@@ -247,32 +243,41 @@ func (p *Protocol) processFork(fork *chainmanager.Fork) (anchorBlockIDs iotago.B
 		}
 		mainChainCommitment := mainChainChainCommitment.Commitment()
 
-		// TODO: should we retry?
-		p.networkProtocol.RequestAttestations(fork.ForkedChain.Commitment(i).ID(), fork.Source)
-
-		select {
-		case result := <-ch:
-			if result.err != nil {
-				return nil, false, true, ierrors.Wrapf(result.err, "failed to verify commitment %s", result.commitment.ID())
-			}
-
-			anchorBlockIDs = append(anchorBlockIDs, result.blockIDs...)
-
-			// Count how many consecutive slots are heavier/lighter than the main chain.
-			switch {
-			case result.actualCumulativeWeight > mainChainCommitment.CumulativeWeight():
-				heavierCount++
-			case result.actualCumulativeWeight < mainChainCommitment.CumulativeWeight():
-				heavierCount--
-			default:
-				heavierCount = 0
-			}
-
-			if decided, doSwitch := forkChoiceRule(heavierCount); decided {
-				return anchorBlockIDs, doSwitch, false, nil
+		ticker := time.NewTicker(p.optsAttestationRequesterTryInterval)
+		defer ticker.Stop()
```

Comment:
As the `defer` is function scoped, it might actually not be properly called in this setup right now. I'd recommend putting everything from the ticker and the loop into its own function. 

This should also allow you to get rid of the `received` variable but instead just return the result or an error. 
The switch should still be evaluated on this level so that the function really just does request (potentially multiple) times and waits for the response without having actual logic in it